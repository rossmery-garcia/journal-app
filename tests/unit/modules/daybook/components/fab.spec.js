import { shallowMount } from '@vue/test-utils'
import Fab from '@/modules/daybook/components/Fab'

describe('Fab component', () => {

  test('Should match with the snapshot', () => {
    const wrapper = shallowMount( Fab )

    expect( wrapper.html() ).toMatchSnapshot()
  });

  test('Should show the default icon', () => {

    const wrapper = shallowMount( Fab )
    const iTag = wrapper.find('i')

    expect( iTag.classes('fa-plus') ).toBeTruthy()
  })

  test('Should show the argument icon: fa-circle', () => {
    const wrapper = shallowMount( Fab, {
      props: {
        icon: 'fa-circle'
      }
    })

    const iTag = wrapper.find('i')

    expect( iTag.classes('fa-circle') ).toBeTruthy()
  })

  test('Should emit the on:click event when clicked', () => {
    const wrapper = shallowMount( Fab )

    wrapper.find('button').trigger('click')

    expect(wrapper.emitted('on:click')).toHaveLength(1)
  })
})