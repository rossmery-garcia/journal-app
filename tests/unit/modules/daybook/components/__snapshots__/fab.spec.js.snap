// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab component Should match with the snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
